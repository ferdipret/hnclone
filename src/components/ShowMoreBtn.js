import React from 'react'
import '../styles/button.css'

function ShowMoreBtn({ fetchMoreStories, storiesLeft }) {
  return (
    storiesLeft > 0 && (
      <button
        className="show-more-btn"
        type="action"
        onClick={fetchMoreStories}>
        SHOW MORE ({storiesLeft})
      </button>
    )
  )
}

export default ShowMoreBtn
