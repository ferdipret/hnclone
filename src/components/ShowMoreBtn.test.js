import React from 'react'
import { mount } from 'enzyme'
import ShowMoreBtn from './ShowMoreBtn'

describe('ShowMoreBtn', () => {
  const props = {
    storiesLeft: 100,
  }
  const showMoreBtn = mount(<ShowMoreBtn {...props} />)

  it('renders properly', () => {
    expect(showMoreBtn).toMatchSnapshot()
  })

  describe('if the `storiesLeft` prop is greater than 0', () => {
    it('should render the button', () => {
      expect(showMoreBtn.props().storiesLeft).toEqual(100)
      expect(showMoreBtn.find('.show-more-btn').exists()).toBe(true)
    })
  })

  describe('if the `storiesLeft` props is less than or equal to 0', () => {
    it('should not render the button', () => {
      showMoreBtn.setProps({ storiesLeft: 0 })

      expect(showMoreBtn.props().storiesLeft).toEqual(0)
      expect(showMoreBtn.find('.show-more-btn').exists()).toBe(false)
    })
  })
})
