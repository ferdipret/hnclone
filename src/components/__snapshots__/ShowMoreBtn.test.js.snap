// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ShowMoreBtn renders properly 1`] = `
ReactWrapper {
  "length": 1,
  Symbol(enzyme.__unrendered__): <ShowMoreBtn
    storiesLeft={100}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "storiesLeft": 100,
    },
    "ref": null,
    "rendered": Object {
      "instance": <button
        class="show-more-btn"
        type="action"
>
        SHOW MORE (
        100
        )
</button>,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "SHOW MORE (",
          100,
          ")",
        ],
        "className": "show-more-btn",
        "onClick": undefined,
        "type": "action",
      },
      "ref": null,
      "rendered": Array [
        "SHOW MORE (",
        "100",
        ")",
      ],
      "type": "button",
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "storiesLeft": 100,
      },
      "ref": null,
      "rendered": Object {
        "instance": <button
          class="show-more-btn"
          type="action"
>
          SHOW MORE (
          100
          )
</button>,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "SHOW MORE (",
            100,
            ")",
          ],
          "className": "show-more-btn",
          "onClick": undefined,
          "type": "action",
        },
        "ref": null,
        "rendered": Array [
          "SHOW MORE (",
          "100",
          ")",
        ],
        "type": "button",
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
